### ISSUE TITLE
Parentheses in string literals can cause errors

### ISSUE THREAD

**Issue by alangpierce** [decaffeinate/decaffeinate#270]
Here's a small test case demonstrating this:
```
a(')').b
```
http://decaffeinate.github.io/decaffeinate/repl/#?evaluate=true&code=a(')').b

It gives the error "cannot find '.' in member access".

That was the problem I specifically ran into, but an even simpler case that fails is just `a(')')`, which produces this incorrect intermediate JS (using some improved error logging I have set up):
```
an intermediate stage failed to parse: Unexpected token (1:5)
> 1 | a(')';)
    |      ^
  2 | 
```

**Comment by alangpierce** [comment]
Looks like this was fixed by the change to trimNonMatchingParentheses.js in 
https://github.com/decaffeinate/decaffeinate-parser/commit/2684933f9ab44ca725542dc1957e30721c564ab1 , so just waiting on a release of decaffeinate-parser and then decaffeinate.

**Comment by eventualbuddha** [comment]
[Fixed in v2.17.2](http://decaffeinate-project.org/repl/#?evaluate=false&code=if%20a%20%26%26%20%28b%3D%3D'%29'%29%0A%20%20x).
